src/pint/models/parameter.py

```diff
@@ -1581,19 +1593,20 @@ def select_toa_mask(self, toas):
         # get the table columns
         # TODO Right now it is only supports mjd, freq, tel, and flagkeys,
         # We need to consider some more complicated situation
+        # THis will have a problem if the key has '-' in the middle of the key.
```

Comment:
This is a bug! It's easy enough to remove it only at the beginning:
```python
if self.key.startswith("-"):
    key = self.key[1:]
else:
    key = self.key
```
for example.

Reply:
Changed.